Update CONTRIBUTING.md code structure section

elomscansio opened the pull request:
The CONTRIBUTING.md file contains outdated information in the section describing the code structure which I updated to reflect the current code strucuture.

ffmancera reviewed (changes requested):
While I agree with removing these items, the PR is incomplete. There are new directories like `src/`, `src/rust/.. and others. It would be really useful to have them documented.

If you have questions about them, feel free to ask us.

cathay4t reviewed (changes requested):
Besides removing outdated lines, please also include doc for new code layout:

 * `rust/src/lib` for rust crate.
 * `rust/src/cli` for `nmstatectl` command line tool.
 * etc

elomscansio committed:
Update code structure section in CONTRIBUTING.md (#2592)

The CONTRIBUTING.md file had outdated information regarding the code structure,
which has now been updated to accurately reflect the current structure.

Signed-off-by: elomscansio <163124154+elomscansio@users.noreply.github.com>

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,18 @@ serving the tests of Nmstate.
 querying kernel network status. 
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
+- `./rust/src/lib/nm/` Contains the nmstate NetworkManager plugin, currently for 
 querying network status and applying network states. 
```
NetworkManager is not a plugin anymore. A better definition would be `Contains the code related to NetworkManager communication and structures.`

elomscansio replied:
ok, Thanks

elomscansio replied:
done

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,18 @@ serving the tests of Nmstate.
 querying network status and applying network states. 
 
-- `./libnmstate/plugins/` Contains the plugins supported by default (e.g. ovsdb). 
+- `./rust/src/lib/ovsdb/` Contains ovsdb plugin supported by default. 
```
Same here. We do not have plugins anymore. A better definition would be `Contains the code related to ovsdb communication and structures`

elomscansio replied:
ok

elomscansio replied:
changed

elomscansio commented:
I have modified the CONTRIBUTE.md to include new structure

ffmancera reviewed (commented):
There are still missing directories, please add them too.

In addition, remember to sign your commit with `-s`

ffmancera commented:
`There are still missing directories, please add them too.`

This comment was not addressed.

elomscansio commented:
pls can i get a hint desc of the following remaining dir so i can add them to the project structure section of the [CONTRIBUTE.md](https://github.com/nmstate/nmstate/pull/2597/commits/e934c85ff07fcd2c0257745072a6c9c0faecd643) file

```
./logo - Project logos
./k8s - 
./rust/src/clib - Contain Nmstate C binding files
./rust/src/go/nmstate - 
./rust/src/python/libnmstate - 
./rust/src/python - 
```

liangwen12year commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,17 @@ serving the tests of Nmstate.
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
-querying network status and applying network states. 
+- `./rust/src/lib/nm/` Contains the code related to NetworkManager communication and structures.
```
Contains the code related to applying the settings via NetworkManager backend?

cathay4t commented:
> pls can i get a hint desc of the following remaining dir so i can add them to the project structure section of the [CONTRIBUTE.md](https://github.com/nmstate/nmstate/pull/2597/commits/e934c85ff07fcd2c0257745072a6c9c0faecd643) file
> 

> ./logo - Project logos

Logos used for publication.

> ./k8s - 

k8s folder holds the scripts for kubernetes-nmstate CI

> ./rust/src/clib - Contain Nmstate C binding files

C binding of nmstate written in rust

> ./rust/src/go/nmstate - 

Go binding of nmstate written in go wrapping the C binding of nmstate

> ./rust/src/python/libnmstate - 
> ./rust/src/python - 

Python library of nmstate written in python wrapping the C binding of nmstate

elomscansio commented:
> > pls can i get a hint desc of the following remaining dir so i can add them to the project structure section of the [CONTRIBUTE.md](https://github.com/nmstate/nmstate/pull/2597/commits/e934c85ff07fcd2c0257745072a6c9c0faecd643) file
> 
> > ./logo - Project logos
> 
> Logos used for publication.
> 
> > ./k8s -
> 
> k8s folder holds the scripts for kubernetes-nmstate CI
> 
> > ./rust/src/clib - Contain Nmstate C binding files
> 
> C binding of nmstate written in rust
> 
> > ./rust/src/go/nmstate -
> 
> Go binding of nmstate written in go wrapping the C binding of nmstate
> 
> > ./rust/src/python/libnmstate -
> > ./rust/src/python -
> 
> Python library of nmstate written in python wrapping the C binding of nmstate

Thanks 

how about 
./rust/src/lib/policy - Contains code related to enforcing network policy 
./rust/src/lib/statistic, ./rust/src/lib/statistic/feature - Contains code for statistic
./rust/src/lib/query_apply - Contains code that applies to commands
./rust/src/lib/revert - 
./rust/src/lib/unit_tests - Contains test code for lib

cathay4t commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,28 @@ serving the tests of Nmstate.
 
-- `./libnmstate/nm/` Contains the nmstate NetworkManager plugin, currently for 
-querying network status and applying network states. 
+ - `./rust/src/lib/ifaces/` Contains the classes including the attributes of the 
```
This one is not required

elomscansio replied:
ok, I have removed it

cathay4t reviewed (changes requested):
Please remove `./rust/src/lib/ifaces/`.

Besides that looks good to me.

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
+
+- `./rust/src/clib` C binding of nmstate written in rust
+
+- `./rust/src/go/nmstate` Contains the classes including the attributes of the 
 interfaces.
```
This should be `Go library of nmstate wrapping the C bindings`

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
+
+- `./k8s` k8s folder holds the scripts for kubernetes-nmstate CI
+
+- `./rust/src/clib` C binding of nmstate written in rust
```
`C bindings of nmstate` is enough.

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
 interfaces.
  
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
+- `./rust/src/python/libnmstate`, `./rust/src/python` Python library of nmstate written in python wrapping the C binding of nmstate
```
`rust/src/python Python library of nmstate wrapping the C bindings`

ffmancera commented on CONTRIBUTING.md:
```diff
@@ -24,20 +24,25 @@ serving the tests of Nmstate.
-- `./libnmstate/nispor/` Contains the nmstate nispor plugin, currently for 
+- `./rust/src/python/libnmstate`, `./rust/src/python` Python library of nmstate written in python wrapping the C binding of nmstate
+ 
+- `./rust/src/lib/nispor/` Contains the nmstate nispor plugin, currently for 
```
Plugins do not exist anymore. `Contains the code related to querying the network state with via Nispor`

elomscansio commented:
I have made the requested changes

cathay4t committed:
Merge branch 'base' into update/contribute-md
